src/test/java/org/cactoos/experimental/ThreadsTest.java

```diff
@@ -92,37 +130,106 @@ void containsResults() {
      */
     @Test
     void failsDueToException() {
+        final ExecutorService extor = Executors.newSingleThreadExecutor();
+        try {
+            new Assertion<>(
+                "Must rethrow error",
+                () -> new LengthOf(
+                    new Threads<String>(
+                        extor,
+                        Duration.ofSeconds(1),
+                        () -> {
+                            this.sleep();
+                            return "txt 1";
+                        },
+                        () -> {
+                            throw new IllegalStateException(
+                                "Something went wrong"
+                            );
+                        }
+                    )
+                ).value(),
+                new Throws<>(
+                    // @checkstyle LineLength (1 line)
+                    "java.io.IOException: java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Something went wrong",
+                    UncheckedIOException.class
+                )
+            ).affirm();
+        } finally {
+            extor.shutdownNow();
+        }
+    }
+
+    /**
+     * Execute the tasks concurrently using {@link Threads} when
+     *  {@link ExecutorService} was initiated by {@link Threads} itself.
+     */
+    @RepeatedTest(ThreadsTest.REPETITIONS)
+    void containsValuesWithInlineExecutorService() throws Exception {
         new Assertion<>(
-            "wraps error into CompletionException",
-            () -> new Threads<String>(
-                Executors.newSingleThreadExecutor(),
-                new ScalarOf<>(
+            "Must contain results from the callables",
```

Comment:
@victornoel perhaps we could write : "Must contain results from the callables when using inline executor service" to make a difference with test `containsResults`